### PR TITLE
chore: plugin compatibility infos

### PR DESCRIPTION
**Pull request by alfredotoselli**
# Description

Added min and max version on plugin manifest. 
If we agree on the structure I'll update admin-vue and ts-client (Plugin.ts type)

We want to enforce it? e.g. on plugin installation or activation (Not necessary for me)

Related to issue #960 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by pieroit** [comment]
No enforcing, will make things way more complicated

File: core/cat/mad_hatter/plugin.py [hunk 1]
```diff
@@ -241,6 +241,16 @@ def _load_manifest(self):
         meta["thumb"] = json_file_data.get("thumb", "")
         meta["version"] = json_file_data.get("version", "0.0.1")
 
+        # Core compatibility
+        compatibility = json_file_data.get(
+            "compatibility",
+            {
```
Comment (pieroit) [inline core/cat/mad_hatter/plugin.py:247]: Please keep the structure flat for the moment, maybe with a more explicit naming, i.e.
`version`, ..., `min_cat_version`, `max_cat_version`

File: core/cat/mad_hatter/plugin.py [hunk 2]
```diff
@@ -241,6 +241,16 @@ def _load_manifest(self):
...
+                "min_version": "not specified",
+                "max_version": "not specified",
```
Comment (pieroit) [inline core/cat/mad_hatter/plugin.py:249]: maybe empty string as default value, we made a mess in the past with a different default for every key XD

**Review by pieroit — CHANGES_REQUESTED** [review]
Thanks @alfredotoselli left some change request

**Comment by alfredotoselli** [comment]
Here is [admin-vue PR](https://github.com/cheshire-cat-ai/admin-vue/pull/112)

About an eventual warning we could:
- check the versions on `get_available_plugins` API, log a warning and return the info to be used by admin-vue
- check the version somewhere else in BE, just log a warning 
- check the version and show warning only in admin vue
- not doing it at all

**Comment by pieroit** [comment]
> Here is [admin-vue PR](https://github.com/cheshire-cat-ai/admin-vue/pull/112)
> 
> About an eventual warning we could:
> 
> * check the versions on `get_available_plugins` API, log a warning and return the info to be used by admin-vue
> * check the version somewhere else in BE, just log a warning
> * check the version at FE, showing a warning only in admin vue
> * not doing it at all

Thanks @alfredotoselli, just merged.

Let's see what the contribs think about the strategy here.
Any take of responsibility on plugin compatibility is going to cost maintaining, so my preference is on waiting to better focus and with some feedback at hand

**Pull request merged**